logpolicy/logpolicy.go

```diff
@@ -481,13 +506,10 @@ func NewWithConfigPath(collection, dir, cmdName string, netMon *netmon.Monitor,
 		earlyErrBuf.WriteByte('\n')
 	}
 
-	if dir == "" {
-		dir = LogsDir(earlyLogf)
-	}
-	if cmdName == "" {
-		cmdName = version.CmdName()
-	}
+	dir := cmp.Or(opts.Dir, LogsDir(earlyLogf))
```

Comment:
revert. cmp.Or isn't a macro :) 

LogsDir does a lot of stuff that you're doing here just to throw it away if opts.Dir is non-zero.

Reply:
Ah yes. This made me realize that `cmp.Or` is less useful that I'd like.